packages/lib/zupass-client/src/protocol.ts

```diff
@@ -1,51 +1,91 @@
 import { z } from "zod";
 import { ZappSchema } from "./zapp";
 
+/**
+ * Messages that can be sent directly to a window, before MessagePorts have\
+ * been transferred.
+ */
 export enum WindowMessageType {
-  ZUPASS_CLIENT_CONNECT = "zupass-client-connect"
+  ZUPASS_CLIENT_CONNECT = "zupass-client-connect",
+  ZUPASS_HOST_CONNECT = "zupass-host-connect"
 }
 
+/**
+ * Messages that can be sent via MessagePorts.
+ */
 export enum RPCMessageType {
   ZUPASS_CLIENT_INVOKE = "zupass-client-invoke",
   ZUPASS_CLIENT_INVOKE_RESULT = "zupass-client-invoke-result",
   ZUPASS_CLIENT_INVOKE_ERROR = "zupass-client-invoke-error",
   ZUPASS_CLIENT_READY = "zupass-client-ready",
   ZUPASS_CLIENT_SHOW = "zupass-client-show",
-  ZUPASS_CLIENT_HIDE = "zupass-client-hide"
+  ZUPASS_CLIENT_HIDE = "zupass-client-hide",
+  ZUPASS_CLIENT_CONNECT = "zupass-client-connect"
```

Comment:
this is a dupe of `WindowMessageType.ZUPASS_CLIENT_CONNECT`. any reason we need to have this as well?

Reply:
When Zupass is embedded inside an iframe, this message is sent to the window. However, when Zupass is embedding a Zapp, Zupass initiates a MessagePort and so the message is sent over that channel.

I ought to tidy this up and document it - will do that on Monday.